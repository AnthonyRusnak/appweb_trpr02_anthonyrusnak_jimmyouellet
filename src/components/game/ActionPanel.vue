<script setup lang="ts">
import { ActionType } from '../../scripts/enums';
import { PLAYER_HEAL_COST_PER_AMOUNT, PLAYER_SUMMON_COST, DEFAULT_PLAYER_HEALTH } from '../../scripts/consts';

const props = defineProps({
  playerLifeForce: Number,
  playerHealth: Number,
  isPlaying: Boolean
})

const emit = defineEmits<{
  (event: 'playerAct', action: ActionType): void,
}>()

function act(action : ActionType): void{
  emit('playerAct', action)
}

</script>

<template>
  <div class="container">
    <div class="row align-items-start gx-4">
        <div class="bg-dark border border-success rounded border-4" style="--bs-bg-opacity: .75;">
          <div class="d-flex justify-content-around">
            <button :disabled="isPlaying" @click="act(ActionType.ATTACK)" class="btn btn-dark border border-success border-4 fs-1">
              <img class="icon" src="../../assets/images/necro_sword.png" alt="Attaque" />
            </button>

            <button :disabled="isPlaying" @click="act(ActionType.DEFEND)" class="btn btn-dark border border-success border-4 fs-1">
              <img class="icon" src="../../assets/images/necro_shield.png" alt="DÃ©fense" />
            </button>

            <button :disabled="(playerLifeForce?? 0) < PLAYER_SUMMON_COST || isPlaying" 
            @click="act(ActionType.SUMMON)" class="btn btn-dark border border-success border-4 fs-1">
              <img class="icon" src="../../assets/images/necro_hand.png" alt="Invocation" />
            </button>

            <button :disabled="(playerLifeForce?? 0) < PLAYER_HEAL_COST_PER_AMOUNT || (playerHealth?? 0) >= DEFAULT_PLAYER_HEALTH || isPlaying"
            @click="act(ActionType.HEAL)" class="btn btn-dark border border-success border-4 fs-1">
              <img class="icon" src="../../assets/images/necro_heal.png" alt="Soin" />
            </button>

            <button :disabled="isPlaying" @click="act(ActionType.FLEE)" class="btn btn-dark border border-success border-4 fs-1">
              <img class="icon" src="../../assets/images/necro_flee.png" alt="Fuir" />
            </button>
          </div>
        </div>
    </div>
  </div>
</template>

<style scoped>
.icon {
  width: 5rem;
  height: 5rem;
}
</style>