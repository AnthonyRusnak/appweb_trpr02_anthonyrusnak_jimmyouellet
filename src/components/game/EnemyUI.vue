<script setup lang="ts">
import type Character from '../../scripts/interfaces'
import CharacterName from './CharacterName.vue';
import CharacterStats from './CharacterStats.vue';
import CharacterHealthBar from './CharacterHealthBar.vue';
import { ENEMY_COLOR } from '../../scripts/consts';

const props = defineProps({
    enemy: Object as () => Character,
    enemyCurrentHealth: Number,
})

</script>

<template>
    <div class="bg-dark border border-success rounded border-4" style="--bs-bg-opacity: .90;">
        <div class="container">
            <CharacterName :characterName="'Nicolas Cage'" :characterTitle="'Casseur de mort'" :color="ENEMY_COLOR" />
            <div class="row align-items-end pt-4 pb-2">
                <CharacterHealthBar :characterHealthPercentage="100" :characterCurrentHealth="100" :maxHealth="100" :color="ENEMY_COLOR" />
                <CharacterStats :lifeForce="100" :experience="2" :strength="100" :color="ENEMY_COLOR" />
            </div>
        </div>
    </div>
</template>