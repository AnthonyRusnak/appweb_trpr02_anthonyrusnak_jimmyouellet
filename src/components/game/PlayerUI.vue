<script setup lang="ts">
import { computed, type Ref, ref, watch, type ComputedRef } from 'vue';
import type Character from '../../scripts/interfaces';
import CharacterName from './CharacterName.vue';
import CharacterStats from './CharacterStats.vue';
import CharacterHealthBar from './CharacterHealthBar.vue';
import { PLAYER_COLOR } from '../../scripts/consts';

const props = defineProps({
    player: Object as () => Character,
    playerCurrentHealth: Number,
    playerMinionCount: Number,
})

const playerHealthPercentage: Ref<number> = computed(() => ((props.playerCurrentHealth??0) / props.player!.vitality) * 100)

</script>
<template>
    <div class="bg-dark border border-success rounded border-4" style="--bs-bg-opacity: .90;">
        <div class="container">
          <CharacterName :characterName="player?.name" :characterTitle="player?.title" :color="PLAYER_COLOR" />
            <div class="row align-items-end pt-4 pb-2">
              <CharacterStats :lifeForce="player?.lifeForce" :experience="player?.experience" 
              :minionCount="playerMinionCount" :strength="player?.strength" :color="PLAYER_COLOR" />
              <CharacterHealthBar :characterHealthPercentage="playerHealthPercentage" :characterCurrentHealth="playerCurrentHealth" :color="PLAYER_COLOR" />
            </div>
        </div>
    </div>
</template>

<style scoped>
</style>