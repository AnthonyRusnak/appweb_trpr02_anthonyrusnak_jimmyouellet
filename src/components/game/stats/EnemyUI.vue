<script setup lang="ts">
import {type Character} from '../../../scripts/interfaces'
import CharacterName from '../stats/CharacterName.vue';
import CharacterStats from './CharacterStats.vue';
import CharacterHealthBar from './CharacterHealthBar.vue';
import { ENEMY_COLOR } from '../../../scripts/consts';

const props = defineProps({
    enemy: Object as () => Character,
    enemyCurrentHealth: Number,
})
</script>

<template>
    <div class="bg-dark border border-success rounded border-4 characterUI container-fluid" style="--bs-bg-opacity: .90;">
        <div class="text-end">
            <CharacterName :characterName="enemy?.name" :characterTitle="enemy?.title" :color="ENEMY_COLOR" />
        </div>
        <div class="row align-items-end pt-4 pb-2">
            <CharacterHealthBar :characterMaxHealth="enemy?.vitality" :characterCurrentHealth="enemyCurrentHealth" :color="ENEMY_COLOR" />
            <CharacterStats :lifeForce="enemy?.lifeForce" :experience="enemy?.experience" :strength="enemy?.strength" :color="ENEMY_COLOR" />
        </div>
    </div>
</template>