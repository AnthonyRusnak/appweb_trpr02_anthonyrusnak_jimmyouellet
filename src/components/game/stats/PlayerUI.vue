<script setup lang="ts">
import { defineProps } from 'vue';
import type Character from '../../../scripts/interfaces';
import CharacterName from '../stats/CharacterName.vue';
import CharacterStats from './CharacterStats.vue';
import CharacterHealthBar from './CharacterHealthBar.vue';
import { PLAYER_COLOR } from '../../../scripts/consts';

const props = defineProps({
    player: Object as () => Character,
    playerCurrentHealth: Number,
    playerMinionCount: Number,
})

</script>
<template>
    <div class="bg-dark border border-success rounded border-4 characterUI container-fluid" style="--bs-bg-opacity: .90;">
      <CharacterName :characterName="player?.name" :characterTitle="player?.title" :color="PLAYER_COLOR" />
        <div class="row align-items-end pt-4 pb-2">
          <CharacterStats :lifeForce="player?.lifeForce" :experience="player?.experience" 
          :minionCount="playerMinionCount" :strength="player?.strength" :color="PLAYER_COLOR" />
          <CharacterHealthBar :characterMaxHealth="player?.vitality" :characterCurrentHealth="playerCurrentHealth" :color="PLAYER_COLOR" />
        </div>
    </div>
</template>

<style scoped>
</style>