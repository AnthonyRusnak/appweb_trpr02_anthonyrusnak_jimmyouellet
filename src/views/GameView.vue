<script setup lang="ts">
import { onMounted, ref, defineProps, type Ref, computed, h } from 'vue'
import { gameService } from '../services/gameService'
import { useToast } from 'vue-toast-notification'
import { type Title, type Undead, type Character, type PopUpText } from '../scripts/interfaces'
import Loading from 'vue-loading-overlay'
import 'vue-loading-overlay/dist/css/index.css'
import 'vue-loading-overlay/dist/css/index.css'
import 'vue-toast-notification/dist/theme-sugar.css'
import GameLogger from '../components/game/GameLogger.vue'
import ActionPanel from '../components/game/ActionPanel.vue'
import EnemyUI from '../components/game/stats/EnemyUI.vue'
import PlayerUI from '../components/game/stats/PlayerUI.vue'
import { ActionType, EventType } from '../scripts/enums'
import { DEFAULT_PLAYER_EXPERIENCE, DEFAULT_PLAYER_HEALTH, DEFAULT_PLAYER_LIFE_FORCE, DEFAULT_PLAYER_STRENGTH, ENEMY_COLOR, ENEMY_MINION_KILL_CHANCE, GAME_COLOR, GAME_DELAY, PLAYER_BLOCK_DAMAGE_DIVIDER, PLAYER_COLOR, PLAYER_HEAL_AMOUNT, PLAYER_HEAL_COST_PER_AMOUNT, PLAYER_SUMMON_COST, UNDEAD_COLOR, POP_UP_FLEE, POP_UP_LOSS, POP_UP_FIGHT_OVER, POP_UP_PAGE_EXIT, GAME_FIGHT_AMOUNT, POP_UP_WIN } from '@/scripts/consts'
import { handleAttack, handleHealthAjustment, isCharacterDead, delay } from '../scripts/utils/gameUtils'
import PopUp from '../components/game/PopUp.vue'
import { useRouter } from 'vue-router'

const router = useRouter()
const allowLeave: Ref<boolean> = ref(false)

// Database elements
const undeads: Ref<Undead[]> = ref([])
const enemies: Ref<Character[]> = ref([])

// Player elements
const player: Ref<Character> = ref({} as Character)

const playerHealth: Ref<number> = ref(DEFAULT_PLAYER_HEALTH)
const playerIsBlocking: Ref<boolean> = ref(false)
let playerIsFleeing: boolean = false
const playerMinions: Ref<Undead[]> = ref([])
const playerMinionCount: Ref<number> = computed(() => playerMinions.value.length)

// Enemy elements
const currentEnemy: Ref<Character> = ref({} as Character)
const currentEnemyHealth: Ref<number> = ref(0)
let currentEnemyIsDead: boolean = false

// Game elements
const gameLog: Ref<string[][]> = ref([])
const currentFight: Ref<number> = ref(1)
const isLoading: Ref<boolean> = ref(false)
const isTurnPlaying: Ref<boolean> = ref(false)

// PopUp elements
const isPopUpShowing: Ref<boolean> = ref(false)
const popUpEventType: Ref<string> = ref('')
const popUpButtonAction: Ref<Function> = ref(() => {})
const currentPopUpText: Ref<PopUpText> = ref({} as PopUpText)

const props = defineProps({
  playerName: String,
  playerTitleId: String,
})

async function act(action: ActionType): Promise<void> {
  isTurnPlaying.value = true

  playerAct(action)
  await delay(GAME_DELAY)

  if (!playerIsFleeing){
    if (!currentEnemyIsDead ) {
      enemyAct()
      await delay(GAME_DELAY)

      undeadsAct()
      await delay(GAME_DELAY * playerMinionCount.value)
    }

    if (currentEnemyIsDead) {
      handleFightOver()
    } 
    gameLog.value.push([`-----------------------------------------------------`, GAME_COLOR])
  }
  isTurnPlaying.value = false
}

//PLAYER ACTIONS SECTION
function playerAct(action: ActionType): void {
  playerIsBlocking.value = false
  playerIsFleeing = false
  switch (action) {
    case ActionType.ATTACK:
      attack()
      break
    case ActionType.DEFEND:
      defend()
      break
    case ActionType.HEAL:
      heal()
      break
    case ActionType.SUMMON:
      summon()
      break
    case ActionType.FLEE:
      flee()
      break
    default:
      break
  }
}
function attack(): void {
  const damage = handleAttack(player.value.experience, player.value.strength)
  if (damage === 0) {
    gameLog.value.push([`Votre attaque a raté.` , PLAYER_COLOR])
  }else{
    currentEnemyHealth.value -= damage
    handleHealthAjustment(currentEnemyHealth)
    handleEnemyDeath()
    gameLog.value.push([`Vous lancez des flammes nécrotiques pour ${damage} points de dégâts.` , PLAYER_COLOR])
  }
}
function defend(): void {
  playerIsBlocking.value = true
  gameLog.value.push([`Vous formez un voile nécrotique autour de vous.` , PLAYER_COLOR])
}
function heal(): void{
  const preHealHealth: number = playerHealth.value;
  let healCost: number = 0

  while (player.value.lifeForce >= PLAYER_HEAL_COST_PER_AMOUNT && playerHealth.value < player.value.vitality) {
    player.value.lifeForce -= PLAYER_HEAL_COST_PER_AMOUNT
    playerHealth.value += PLAYER_HEAL_AMOUNT
    healCost += PLAYER_HEAL_COST_PER_AMOUNT
  }
  if (playerHealth.value >= player.value.vitality) {
    playerHealth.value = player.value.vitality
  }

  let healAmount: number = playerHealth.value - preHealHealth
  gameLog.value.push([`Vous abosrbez ${healCost} points de force vitale pour vous soigner de ${healAmount} points de vie.`, PLAYER_COLOR])
}
function summon(): void {
  player.value.lifeForce -= PLAYER_SUMMON_COST
  let newMinion: Undead = undeads.value[Math.floor(Math.random() * undeads.value.length)]
  playerMinions.value.push(newMinion)
  gameLog.value.push([`Vous réanimez un(e) ${newMinion.type} pour servir votre cause.`, PLAYER_COLOR])
}
function flee(): void{
  playerIsFleeing = true
  setPopUp(EventType.FLEE, POP_UP_FLEE, escape)
}

function escape(healing: boolean): void{
  if (healing) heal()
  getNextFight()
}


//ENEMY ACTIONS SECTION
function enemyAct(): void {
  if (playerMinionCount.value > 0 && Math.random() < ENEMY_MINION_KILL_CHANCE) {
    enemyKillMinion()
  } else {
    enemyAttack()
  }
}

function enemyAttack(): void {
  let damage = handleAttack(currentEnemy.value.experience, currentEnemy.value.strength)
  if (damage === 0) {
    gameLog.value.push([`${currentEnemy.value.name} rate son attaque.`, ENEMY_COLOR])
  } 
  else {
    if (playerIsBlocking.value){
      damage = Math.floor(damage / PLAYER_BLOCK_DAMAGE_DIVIDER)
    }
    playerHealth.value -= damage
    handleHealthAjustment(playerHealth)
    handlePlayerDeath()
    gameLog.value.push([`${currentEnemy.value.name} vous attaque pour ${damage} points de dégâts.`, ENEMY_COLOR])
  }
}

function enemyKillMinion(){
  let minion = playerMinions.value[Math.floor(Math.random() * playerMinions.value.length)]
  playerMinions.value.splice(playerMinions.value.indexOf(minion), 1)
  gameLog.value.push([`${currentEnemy.value.name} à éliminé votre ${minion.type}.`, ENEMY_COLOR])
}

//UNDEAD ACTIONS SECTION
function undeadsAct(): void {
  let currentMinion = 0
  playerMinions.value.forEach(async minion => {
    currentMinion++
    await delay(GAME_DELAY * (currentMinion-1))
    undeadAttack(minion)
  })
}
function undeadAttack(minion: Undead): void {
  let damage = handleAttack(minion.experience, minion.strength)
  if (damage === 0) {
    gameLog.value.push([`${minion.type} rate son attaque.`, UNDEAD_COLOR])
  } else {
    currentEnemyHealth.value -= damage
    handleHealthAjustment(currentEnemyHealth)
    handleEnemyDeath()
    gameLog.value.push([`${minion.type} attaque pour ${damage} points de dégâts.` , UNDEAD_COLOR])
  }
}

//GAME SECTION
function handlePlayerFightWin(healing: boolean): void {
  if (healing) heal()
  getNextFight()
}

function getNextFight(): void {
  prepNextFight()
  getNewEnemy()
  gameLog.value.push([`${currentEnemy.value.name} se dresse devant vous.`, GAME_COLOR])
}

function prepNextFight(): void {
  enemies.value.splice(enemies.value.indexOf(currentEnemy.value), 1)
  gameLog.value = [];
  currentFight.value++
}

function getNewEnemy(): void {
  currentEnemyIsDead = false
  currentEnemy.value = enemies.value[Math.floor(Math.random() * enemies.value.length)]
  currentEnemyHealth.value = currentEnemy.value.vitality
}

function handleEnemyDeath(): void {
  currentEnemyIsDead = isCharacterDead(currentEnemyHealth.value)
}

function handlePlayerDeath(): void {
  if (isCharacterDead(playerHealth.value)) {    
    setPopUp(EventType.LOSS, POP_UP_LOSS, endGame)
  }
}

function endGame(restart: boolean): void {
  if (restart) {
    location.reload()
  } else {
    router.push({ name: 'Home' })
  }
}

async function winGame(saveScore: boolean): Promise<void> {
  if (saveScore) {
    isLoading.value = true
    let rankId: number = await gameService.getNextRankingId()
    await gameService.addRanking({ id: rankId , name: player.value.name, score: player.value.lifeForce })
    isLoading.value = false
    allowLeave.value = true
    router.push({ name: 'Scores' })
  } else {
    router.push({ name: 'Home' })
  }
}

function handleFightOver(): void {
  player.value.lifeForce += currentEnemy.value.lifeForce
  if (currentFight.value >= GAME_FIGHT_AMOUNT){
    setPopUp(EventType.WIN, POP_UP_WIN, winGame)
  } else{
    setPopUp(EventType.FIGHT_OVER, POP_UP_FIGHT_OVER, handlePlayerFightWin)
  }
}

function setPopUp(eventType: EventType, popUpText: PopUpText, buttonAction: Function): void {
  popUpEventType.value = eventType as string
  currentPopUpText.value = popUpText
  popUpButtonAction.value = buttonAction
  isPopUpShowing.value = true
}

function popUpAct(condition: boolean): void {
  isPopUpShowing.value = false
  popUpButtonAction.value(condition)
}

router?.beforeEach((to) => {
  if (!allowLeave.value){
    setPopUp(EventType.QUIT, POP_UP_PAGE_EXIT, (condition: boolean) => {
      if (condition) {
        allowLeave.value = true
        router.push(to)
      }
    })
  }
  return allowLeave.value
})

onMounted(async () => {
  isLoading.value = true

  try 
  {
    var playerTitle: Title = await gameService.getTitleById(parseInt(props.playerTitleId?? '0'))
    enemies.value = await gameService.getCharacters()
    undeads.value = await gameService.getUndeads()
    getNewEnemy()

    player.value = {
      id: 0,
      name: props.playerName?? '',
      title: playerTitle.name,
      lifeForce: DEFAULT_PLAYER_LIFE_FORCE,
      strength: DEFAULT_PLAYER_STRENGTH,
      experience: DEFAULT_PLAYER_EXPERIENCE,
      vitality: DEFAULT_PLAYER_HEALTH
    } as Character
  } 
  catch (error) 
  {
    useToast().error(
      `Erreur avec le service: ${(error as Error).message}. Veuillez réessayer plus tard.`,
      { duration: 6000 }
    )
  } 
  finally 
  {
    isLoading.value = false
  }
})
</script>

<template>
  <PopUp :isShowing="isPopUpShowing" :eventType="popUpEventType" :popUpText="currentPopUpText" @[popUpEventType]="popUpAct" />
  <template v-if="!isLoading">
    <div class="container overflow-hidden text-light">
      <div class="row align-items-start gx-4">
        <div class="col-4">
          <PlayerUI :player="player" :playerCurrentHealth="playerHealth" :playerMinionCount="playerMinionCount" />
        </div>
        <div class="col-4">
          <GameLogger :gameLog="gameLog" :currentFight="currentFight" />
        </div>
        <div class="col-4">
          <EnemyUI :enemy="currentEnemy" :enemyCurrentHealth="currentEnemyHealth"/>
        </div>
      </div>
      <ActionPanel :isPlaying="isTurnPlaying" :playerHealth="playerHealth" :playerLifeForce="(player?.lifeForce??0)" @playerAct="act" />
    </div>
  </template>
  <Loading :active="isLoading" />
</template>

<style scoped>
</style>