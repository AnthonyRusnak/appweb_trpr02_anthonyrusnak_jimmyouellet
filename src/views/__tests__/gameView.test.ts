import { describe, expect, it, vi } from 'vitest'
import { flushPromises, mount } from '@vue/test-utils'
import GameView from '../../views/GameView.vue'
import { ActionType } from '../../scripts/enums'
import ActionPanel from '../../components/game/ActionPanel.vue'
import PlayerUI from '../../components/game/stats/PlayerUI.vue'
import EnemyUI from '../../components/game/stats/EnemyUI.vue'

describe('GameView.vue', () => {
   it('Should renders properly', async () => {
        const wrapper = mount(GameView, { props: { playerName: 'Test', playerTitleId: '1' } })
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('Should display the action panel', async () => {
        const wrapper = mount(GameView, { props: { playerName: 'Test', playerTitleId: '1' } })
        expect(wrapper.findComponent(ActionPanel).exists()).toBe(true)
    })

    it('Should display the player UI', async () => {
        const wrapper = mount(GameView, { props: { playerName: 'Test', playerTitleId: '1' } })
        expect(wrapper.findComponent(PlayerUI).exists()).toBe(true)
    })

    it('Should display the enemy UI', async () => {
        const wrapper = mount(GameView, { props: { playerName: 'Test', playerTitleId: '1' } })
        expect(wrapper.findComponent(EnemyUI).exists()).toBe(true)
    })

})