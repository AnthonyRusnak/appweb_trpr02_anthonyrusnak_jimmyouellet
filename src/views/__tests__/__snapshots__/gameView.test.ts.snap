// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GameView.vue > Should renders properly 1`] = `
"<div class="modal fade" data-bs-backdrop="static" tabindex="-1" role="dialog" aria-labelledby="popUpModalTitle" aria-hidden="true">
  <div class="modal-dialog modal-dialog-centered" role="document">
    <div class="modal-content border border-success border-5 bg-dark" style="--bs-bg-opacity: .90;">
      <div class="modal-header border-bottom border-success">
        <h5 class="modal-title text-success display-6 outlined-dark" id="popUpModalTitle"></h5>
      </div>
      <div class="modal-body text-light"></div>
      <div class="modal-footer d-flex justify-content-between border-top border-success"><button type="button" class="btn btn-dark btn-lg border border-success border-4"></button><button type="button" class="btn btn-dark btn-lg border border-success border-4"></button></div>
    </div>
  </div>
</div>
<div class="container overflow-hidden text-light">
  <div class="row align-items-start gx-4">
    <div class="col-4">
      <div class="bg-dark border border-success rounded border-4 characterUI container-fluid" style="--bs-bg-opacity: .90;">
        <div data-v-77942875="" class="row align-items-start pt-2">
          <div data-v-77942875="" class="col">
            <div data-v-77942875="" class="text-success outlined-dark display-6"></div>
            <div data-v-77942875="" class="text-title"></div>
          </div>
        </div>
        <div class="row align-items-end pt-4 pb-2">
          <div data-v-df467c64="" class="col m-3">
            <div data-v-df467c64="" class="exp outlined-dark text-success">Apprenti</div>
            <div data-v-df467c64="" class="label outlined-dark text-success">Force: <span data-v-df467c64=""></span></div>
            <div data-v-df467c64="" class="label outlined-dark text-success">Force vitale: <span data-v-df467c64=""></span></div>
            <div data-v-df467c64="" id="minions" class="label outlined-dark text-success">Serviteurs: <span data-v-df467c64="">0</span></div>
          </div>
          <div data-v-ccd5ed8b="" class="progress blue"><span data-v-ccd5ed8b="" class="progress-left"><span data-v-ccd5ed8b="" class="progress-bar"></span></span><span data-v-ccd5ed8b="" class="progress-right"><span data-v-ccd5ed8b="" class="progress-bar"></span></span>
            <div data-v-ccd5ed8b="" class="progress-value">500 PV</div>
          </div>
        </div>
      </div>
    </div>
    <div class="col-4">
      <div data-v-8f1829de="" class="bg-dark border border-success rounded border-4 text-center mb-4" style="--bs-bg-opacity: .90;">
        <div data-v-8f1829de="" class="border-bottom border-success border-4 fs-2"><span data-v-8f1829de="" class="text-success outlined-dark">Combat :</span> 1 <span data-v-8f1829de="" class="text-success outlined-dark">/</span> 5</div>
        <div data-v-8f1829de="" class="log"></div>
      </div>
    </div>
    <div class="col-4">
      <div class="bg-dark border border-success rounded border-4 characterUI container-fluid" style="--bs-bg-opacity: .90;">
        <div class="text-end">
          <div data-v-77942875="" class="row align-items-start pt-2">
            <div data-v-77942875="" class="col">
              <div data-v-77942875="" class="text-success outlined-dark display-6"></div>
              <div data-v-77942875="" class="text-title"></div>
            </div>
          </div>
        </div>
        <div class="row align-items-end pt-4 pb-2">
          <div data-v-ccd5ed8b="" class="progress blue"><span data-v-ccd5ed8b="" class="progress-left"><span data-v-ccd5ed8b="" class="progress-bar"></span></span><span data-v-ccd5ed8b="" class="progress-right"><span data-v-ccd5ed8b="" class="progress-bar"></span></span>
            <div data-v-ccd5ed8b="" class="progress-value">0 PV</div>
          </div>
          <div data-v-df467c64="" class="col m-3">
            <div data-v-df467c64="" class="exp outlined-dark text-success">Apprenti</div>
            <div data-v-df467c64="" class="label outlined-dark text-success">Force: <span data-v-df467c64=""></span></div>
            <div data-v-df467c64="" class="label outlined-dark text-success">Force vitale: <span data-v-df467c64=""></span></div>
            <div data-v-df467c64="" id="minions" class="hidden label outlined-dark text-success">Serviteurs: <span data-v-df467c64=""></span></div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div data-v-8f93bd20="" class="container">
    <div data-v-8f93bd20="" class="row align-items-start gx-4">
      <div data-v-8f93bd20="" class="bg-dark border border-success rounded border-4" style="--bs-bg-opacity: .75;">
        <div data-v-8f93bd20="" class="d-flex justify-content-around"><button data-v-8f93bd20="" class="btn btn-dark border border-success border-4 fs-1"><img data-v-8f93bd20="" class="icon" src="/src/assets/images/necro_sword.png" alt="Attaque"></button><button data-v-8f93bd20="" class="btn btn-dark border border-success border-4 fs-1"><img data-v-8f93bd20="" class="icon" src="/src/assets/images/necro_shield.png" alt="DÃ©fense"></button><button data-v-8f93bd20="" disabled="" class="btn btn-dark border border-success border-4 fs-1"><img data-v-8f93bd20="" class="icon" src="/src/assets/images/necro_hand.png" alt="Invocation"></button><button data-v-8f93bd20="" disabled="" class="btn btn-dark border border-success border-4 fs-1"><img data-v-8f93bd20="" class="icon" src="/src/assets/images/necro_heal.png" alt="Soin"></button><button data-v-8f93bd20="" class="btn btn-dark border border-success border-4 fs-1"><img data-v-8f93bd20="" class="icon" src="/src/assets/images/necro_flee.png" alt="Fuir"></button></div>
      </div>
    </div>
  </div>
</div>"
`;
